build/utils/npm.go

```diff
@@ -122,6 +118,54 @@ func CalculateDependenciesMap(executablePath, srcPath, moduleId string, npmArgs
 	})
 }
 
+func runNpmLsWithNodeModules(executablePath, srcPath string, npmArgs []string, log utils.Log) (data []byte) {
+	npmArgs = append(npmArgs, "--json", "--all", "--long")
+	data, errData, err := RunNpmCmd(executablePath, srcPath, Ls, npmArgs, log)
+	if err != nil {
+		// It is not mandatory for the function to return this error.
+		log.Error("npm list command failed with error:", err.Error())
+	}
+	if len(errData) > 0 {
+		log.Warn("Some errors occurred while collecting dependencies info:\n" + string(errData))
+	}
+	return
+}
+
+func runNpmLsWithoutNodeModules(executablePath, srcPath string, npmArgs []string, log utils.Log, npmVersion *version.Version) ([]byte, error) {
+	isPackageLockExist, isDirExistsErr := utils.IsFileExists(filepath.Join(srcPath, "package-lock.json"), false)
+	if isDirExistsErr != nil {
+		return nil, isDirExistsErr
+	}
+	if !isPackageLockExist {
+		err := installPackageLock(executablePath, srcPath, npmArgs, log, npmVersion)
+		if err != nil {
+			return nil, err
+		}
+	}
+	npmArgs = append(npmArgs, "--json", "--all", "--long", "--package-lock-only")
+	data, errData, err := RunNpmCmd(executablePath, srcPath, Ls, npmArgs, log)
+	if err != nil {
+		log.Error("npm list command failed with error:", err.Error())
+	}
+	if len(errData) > 0 {
+		log.Warn("Some errors occurred while collecting dependencies info:\n" + string(errData))
+	}
+	return data, nil
+}
+
+func installPackageLock(executablePath, srcPath string, npmArgs []string, log utils.Log, npmVersion *version.Version) error {
+	if npmVersion.AtLeast("6.0.0") {
+		npmArgs = append(npmArgs, "--package-lock-only")
+		// Installing package-lock to generate the dependencies map.
+		_, errData, err := RunNpmCmd(executablePath, srcPath, Install, npmArgs, log)
+		if err != nil || len(errData) > 0 {
+			return errors.New("Some errors occurred while installing package-lock: " + err.Error())
+		}
+		return nil
+	}
+	return errors.New("couldn't install package-lock because npm version is less than 6.0.0")
```

Comment:
users who use npm version less than 6.0.0 need to install node-modules or package-lock by themselves, if they already have node_modules or package-lock in their npm project the code flow will not get here and this will never crash them.